### PR TITLE
[WIP] Dreicer electric field

### LINKED ISSUE #529: Calculate Dreicer electric field
We should have a function that calculates the Dreicer electric field, and include a good physical description of what the Dreicer electric field means and it's significance.

My own preference would be to have a function name that represents the physics of what's happening rather than including someone's name, though I don't have a good idea for what that name might be.

The docstring should also cite the [original paper](https://journals.aps.org/pr/abstract/10.1103/PhysRev.115.238).

Thanks!

### PR DESCRIPTION
**Pull request by StanczakDominik**
Closes #529. 

I'm not quite sure if all the variables are correct (oh, transport theory, you magnificent mess). I found a description in Callen's draft book, and I think I kind of sort of understand the process - but I'll look through the other positions linked in the initial issue and see whether I couldn't perhaps find some numerical comparisons to ensure I'm right here.

### CODE REVIEW

File: plasmapy/transport/collisions.py [hunk 1]
```diff
@@ -1791,3 +1796,99 @@ def coupling_parameter(T,
...
+
+    T : ~astropy.units.Quantity
+        Temperature in units of temperature or energy per particle,
+        which is assumed to be equal for both the test particle and
+        the target particle.
+
+    n_i : ~astropy.units.Quantity
+        The ions density in units convertible to per cubic meter.
+
+    ion_particle : Particle
```
Comment (namurphy) [inline plasmapy/transport/collisions.py]: Perhaps shorten this to just `ion`?

File: plasmapy/transport/collisions.py [hunk 2]
```diff
@@ -1791,3 +1796,99 @@ def coupling_parameter(T,
...
+
```
Comment (namurphy) [inline plasmapy/transport/collisions.py]: Should add `coulomb_log` to the list of parameters.

File: plasmapy/transport/collisions.py [hunk 3]
```diff
@@ -1791,3 +1796,99 @@ def coupling_parameter(T,
...
+    References
+    ----------
+    TODO
+    Callen chapter 2
+    """
+    if coulomb_log is None:
+        coulomb_log = Coulomb_logarithm(T,
+                                        n_i,
+                                        (Particle('e'), ion_particle), method=coulomb_log_method
+                                        )
```
Comment (namurphy) [inline plasmapy/transport/collisions.py]: The part that I was confused about earlier was which two particles to calculate the Coulomb logarithm for.  I think you have this right, though I don't fully understand the derivation.

Comment (StanczakDominik) [reply plasmapy/transport/collisions.py]: I'll check this in a few more sources.

Comment (lemmatum) [reply plasmapy/transport/collisions.py]: pg 242 of the Dreicer paper says "Let us consider electron-ion encounters", so I think this is right...?

File: plasmapy/transport/collisions.py [hunk 4]
```diff
@@ -1794,3 +1799,105 @@ def coupling_parameter(T,
...
+@check_quantity()
+def Dreicer_electric_field(T: u.K,
+                           n_i: 1/u.m**3,
+                           ion: Particle,
+                           coulomb_log: numbers.Real=None,
+                           vTh: u.m / u.s = np.nan * u.m / u.s,
+                           coulomb_log_method: str = "classical",
+                           ) -> u.V / u.m:
+    r"""
+    Calculates the minimum electric field for runaway electrons to occur
```
Comment (namurphy) [inline plasmapy/transport/collisions.py]: Perhaps change `Calculates` to `Calculate` for consistency with other functions

Comment (StanczakDominik) [reply plasmapy/transport/collisions.py]: Done!

File: plasmapy/transport/collisions.py [hunk 5]
```diff
@@ -1794,3 +1799,104 @@ def coupling_parameter(T,
...
+@particle_input
+@check_quantity()
+def Dreicer_electric_field(T: u.K,
+                           n_i: 1/u.m**3,
+                           ion: Particle,
+                           coulomb_log: numbers.Real = None,
+                           vTh: u.m / u.s = np.nan * u.m / u.s,
+                           coulomb_log_method: str = "classical",
+                           ) -> u.V / u.m:
+    r"""
```
Comment (SolarDrew) [inline plasmapy/transport/collisions.py]: Why's this a raw string?

File: plasmapy/transport/collisions.py [hunk 6]
```diff
@@ -1794,3 +1799,104 @@ def coupling_parameter(T,
...
+    ~astropy.units.UnitsWarning
+        If units are not provided, SI units are assumed
+
+    ~plasmapy.utils.RelativityWarning
+        If the input velocity is greater than 5% of the speed of
+        light.
+
+    Notes
+    -----
+    TODO
```
Comment (StanczakDominik) [inline plasmapy/transport/collisions.py]: Right, I guess this shouldn't get merged yet...

File: plasmapy/transport/collisions.py (hunk 5 above)
Comment (StanczakDominik) [reply plasmapy/transport/collisions.py]: For latex formulas in the currently empty notes section - thanks for making me check those!
